controller/volume_controller.go

```diff
@@ -3995,6 +4001,8 @@ func (c *VolumeController) prepareReplicasAndEngineForMigration(v *longhorn.Volu
 				return false, true, nil
 			case longhorn.ReplicaModeRW:
 				currentAvailableReplicas[dataPath] = r
+			case "":
+				log.Warnf("Running replica %v wasn't added to engine, will ignore it and continue migration", r.Name)
```

Comment:
Should we clean up or fail this kind of replica?

Reply:
I do not think we should clean it up. If the migration is reverted, it will be good to still have this replica around, as it will theoretically be much faster to rebuild it then it will be to build a new one.

I'm a bit less sure about whether or not we should fail it. Ultimately, I chose not to, as there is nothing wrong with it and it's nice to have it ready to immediately rebuild if the migration is reverted. In addition, if it remains failed for long enough, I think we will clean it up automatically, which I would prefer not to do for the reason above. (Obviously we expect migrations to be fast, but this issue was only raised in the first place due to very long running migrations.) The downside is that there is an extra replica process running for the length of the migration.

If you feel there is a strong reason to fail it instead, please let me know. We can wait to merge until we are agreed.